Write evaluation section (#3)
drvinceknight/literature-review: sec/review.tex

corroborative support offered by model confirmation is `inherently partial'
given the assumptions made to construct a model. However, this confirmation
process has become the standard response for evaluating computer simulation
models --- choose a metric and dataset (or sets thereof) use these to quantify
the quality of the model either alone or against some competitor(s).

These works indicate that the evaluation of a model should not be purely
mathematical or practical, and as such, it invokes a need for some philosophical
thinking. In~\cite{Parker2010}, the author consolidates the concerns of
practitioners and philosophers regarding model evaluation, defining scientific
models as representational tools constructed to achieve a particular purpose or
goal; earlier and more contemporary works concur with this
definition~\cite{Baumberger2017,Caswell1976,Currie2017}. The author suggests
that model evaluation should be considered as the process of determining whether
a model is \emph{adequate} for a purpose, i.e. whether a model is highly likely
to achieve a goal. The author distinguishes this concept of adequacy from

> individual success where a model achieves a purpose in a particular instance.

Reviewer: Paul and I have a paper where we fitted a phase type distribution to a length of stay. I think the "best" fit had 12 phases (or something like that) but when we then put that distribution in to a healthcare queueing model, the decision that was made was the same for a model with just 4 phases. We argued about the importance of not doing more than was necessary.

Not suggesting this is a necessary reference to include here but it could be of interest: http://orca.cf.ac.uk/25345/

Author: I like this. That's a good example :+1:

Author: Added in 5aaf115

way results in a conclusion which is broadly equivalent to a confirmation
process, indicating little about the true quality or robustness of the method.

So, while these works are highly valuable in expanding the discussion of model
evaluation, without concrete and readily applied methods with which to quantify
their approaches, they are unlikely to be adopted widely. This unlikelihood is
not necessarily the fault of the methods, but rather that (in some respects)
convenience and velocity have surpassed merit in the evaluation of models. In an
increasingly competitive environment where becoming the new `state-of-the-art'
has become the norm, regard for robust, thoroughly evaluated models has
declined. The remainder of this section summarises the established evaluation
processes for clustering algorithms and healthcare models, highlighting their
limitations and potential improvements to the current regime.

\subsubsection{Clustering}\label{subsubsec:clustering_evaluation}

Algorithms for clustering, like many machine learning tasks, fall into the
category of models that are primarily evaluated using confirmation processes. To

> reiterate, a confirmation process provides support for a model based on its

Reviewer: Instead of `To reiterate` can we say: `As discussed in <section ...>, a confirmation ...`

Author: Thoughts on this? The section isn't independently numbered

```suggestion
category of models that are primarily evaluated using confirmation processes. As
discussed at the beginning of this section, a confirmation process provides
support for a model based on its
```

Reviewer: Yeah that's great :+1:

alignment with a set of observed data according to some metric. Often lumped in
with the task of classification, clustering is distinct in that it reveals the
underlying structure of a dataset with knowledge that is fixed from the outset
exclusively. However, the metrics used to evaluate clustering algorithms are
often concerned with the retrieval of information such as accuracy, precision,
and recall~\cite{Manning2008}. Examples of this method of evaluation are
abundant, including many of the cited works from Section~\ref{subsec:clustering}
such as~\cite{Agrawal1998,Aljarah2019,Bakr2015,Cao2009,Huang1998}.

Information retrieval metrics can be considered \emph{external criteria} in that

> they measure an aspect of a model's clustering according to a external labelling

Reviewer: ```suggestion
they measure an aspect of a model's clustering according to external labelling
```

Author: ```suggestion
they measure an aspect of a model's clustering according to an external labelling
```

of the dataset~\cite{Mitchell1997}. By evaluating a clustering algorithm based
on a ground truth, the algorithm is effectively being evaluated as a classifier
--- a method which is trained using that ground truth. Hence, the quality of any

> clustering is predicated on the data generation process, and if a clustering

Reviewer: ```suggestion
of the dataset~\cite{Mitchell1997}. The quality of any
clustering is predicated on the data generation process, and if a clustering
```

I found that sentence a bit heavy, I wonder if this suggestion is as effective.

Author: I agree. Will delete.

machine learning practices have been developed for designing
algorithms~\cite{CorbettDavies2018,Friedler2016,Goelz2019,Kearns2017}. For
clustering, these include the centroid-based method introduced
in~\cite{Chen2019} and the hierarchical method from~\cite{Ahmadian2020}. These
methods tend to treat the data points as agents in the model, and rely on
internal mechanisms that maximise their communal utility. Hence, they are in
some respects self-validating. However, to be comparable, objective metrics are
applied showing that the costs of fairness are small losses in objective quality
measures.

Further to particular metrics --- objective or otherwise --- clustering
evaluation has been generalised to study more intrinsic properties of clustering
instances such as clusterability~\cite{Ackerman2009,Ostrovsky2006,Zhang2001}.
These works often rely on an axiomatic description of clustering (examples of
which include~\cite{Bezdek1978,Jardine1971,Kleinberg2002}).
In~\cite{Kleinberg2002}, the author defines clustering as a function, \(f\),
over a dataset, \(X\), which maps a distance metric, \(d\), also over \(X\) to
some partition, \(C\), of \(X\).

> %i.e. a clustering function can be defined as
> %any \(f: \mathcal D_X \to \mathcal B(X)\) where \(\mathcal D_X\) is the set of
> %all distance metrics over \(X\) and \(\mathcal B(X)\) is the set of all
> %partitions of \(X\).

Reviewer: ```suggestion
% TODO Replace with mathematical definitions if appropriate.
%i.e. a clustering function can be defined as
%any \(f: \mathcal D_X \to \mathcal B(X)\) where \(\mathcal D_X\) is the set of
%all distance metrics over \(X\) and \(\mathcal B(X)\) is the set of all
%partitions of \(X\).
```

The primary result of~\cite{Kleinberg2002} is a theorem
stating that no clustering function exists which satisfies all three of the
following `axiomatic' properties: 

\begin{itemize}
    \item Scale-invariance: the function is insensitive to changes in the unit
        distance
        %for every \(d \in D_X\) and for all \(\alpha > 0\) it follows that
        %\(f(d) = f(\alpha d)\)
    \item Richness: all partitions of the dataset are possible outputs of the
        function (by use of an alternative distance function)
        %for every \(C \in \mathcal B(X)\) there exists some \(d \in \mathcal
        %D_X\) such that \(f(d) = C\)
    \item Consistency: the outputted clustering is unchanged when within-cluster
        distances are decreased and between-cluster distances are increased
        %for every \(C\)-consistent distance metric, \(d' \in \left\{d' \in D_X
        %| d'(x, y) \le d(x, y) \forall x \sim_C y \wedge d'(x, y) \ge d(x, y)

>         %\forall x \nsim_C y\right\}\), it follows that \(f(d) = f(d')\)

Reviewer: ```suggestion
       % TODO Replace with mathematical definitions if appropriate.
        %for every \(d \in D_X\) and for all \(\alpha > 0\) it follows that
        %\(f(d) = f(\alpha d)\)
    \item Richness: all partitions of the dataset are possible outputs of the
        function (by use of an alternative distance function)
        % TODO Replace with mathematical definitions if appropriate.
        %for every \(C \in \mathcal B(X)\) there exists some \(d \in \mathcal
        %D_X\) such that \(f(d) = C\)
    \item Consistency: the outputted clustering is unchanged when within-cluster
        distances are decreased and between-cluster distances are increased
        % TODO Replace with mathematical definitions if appropriate.
        %for every \(C\)-consistent distance metric, \(d' \in \left\{d' \in D_X
        %| d'(x, y) \le d(x, y) \forall x \sim_C y \wedge d'(x, y) \ge d(x, y)
        %\forall x \nsim_C y\right\}\), it follows that \(f(d) = f(d')\)
```

Just to document this tweak.

\end{itemize}

However, small changes to these properties accommodate various well-known
clustering algorithms, including single-linkage and \(k\)-means clustering
(discussed in Section~\ref{subsec:clustering}). The authors
of~\cite{BenDavid2008} expand on these axioms to derive necessary and desirable
properties of cluster-quality measures that are entirely agnostic to the
clustering function at hand. The closing sections of that work offer a number of
exemplar cluster-quality measures based on loss, cluster centres and linkage,
including measures that correspond to the all-cluster normalised mean-squared
error, and a within-versus-between variability measure. The latter can be
considered as a generalisation of the silhouette coefficient with respect to its
loss function or distance measure.

> % TODO do we need a brief discussion of datasets? Or should we leave that for

Reviewer: Are you thinking about biases in training data?

Author: Not exclusively. I'm not entirely sure what I'd want to discuss here (thinking for the chapter) as I go into the issues with benchmarks and dataset reliability in the EDO chapter.

For the paper, I think there should be *something* about datasets here.

Reviewer: Sure, I'm not entirely sure what you'd have here but in healthcare a common stated "problem" is lack of data for validation of models. That could perhaps go here (as well as bias in data)

Author: That is something I am going to touch on the last section (healthcare evaluation). Something along the lines of:

- an important point to raise about evaluation of healthcare models are their longterm effects in terms of trust
- point at some of the ethics papers specifically about trustworthiness, and the growing call for unity in best practices for healthcare models (academic and clinical)
- as part of that, one of the key stakeholders in healthcare is the patient (link back to patient-centred care) and so building trust for patients is essential for adopting data-driven decision making in modern healthcare systems
- a crucial component of the patient-system trust relationship regards patient privacy (I have a reference about GDPR in healthcare, and I think there's an article somewhere about Google DeepMind accidentally leaking health data)
- to circumvent these issues, pseudo-privacy can be obtained through data synthesis (as well as anonymisation)
- add on lack of data for assessment, "two birds one stone"-style

Reviewer: That sounds good, those bullet points with a bit of flesh here and there are probably exactly what you need :+1:

downfall of an initiative, such as with the NHS England \emph{care.data}
project~\cite{Carter2015}.

A workaround to the potential misuse of patient data is to synthesise new
datasets from existing, confidential data. The methods by which data synthesis
is achieved are various, but include bespoke, often statistical
models~\cite{Dahmen2019,Dube2014,McLachlan2016,Tucker2020} and, recently,
generative adversarial networks (GANs)~\cite{Avino2018,Torfi2020}. Using
synthetic datasets provides another level of anonymity to the patients, but

> their efficacy against real data has been contested. In a recent

Reviewer: I'm not sure what `efficacy against real data` means here. Would something like: 

```suggestion
their but their validity as representations of real data has been contested. In a recent
```

work?

Author: That's much clearer, yes.
